quinn-proto/src/connection.rs

```diff
@@ -2399,13 +2404,6 @@ impl State {
         })
     }
 
-    pub fn is_handshake(&self) -> bool {
-        match *self {
-            State::Handshake(_) => true,
-            _ => false,
-        }
-    }
-
```

Comment:
Unlike the new `Connection::is_handshaking` method, this can be called when a mutable borrow to some other part of the `Connection` is live. For that reason it might be nice to retain, with the public function being a simple wrapper. On the other hand, we can always bring it back if/when we eventually want to do that, so I don't feel strongly.

Reply:
From the outside it doesn't matter; from the inside I don't think calling `is_handshaking()` will cause more issues? It doesn't take a borrow, just synthesizes a `bool` out of it.

I think I'm angling to remove the final bits of state from `State` variants, too, and just inline them into `Connection` -- though I have a feeling `rem_cid_set` is probably redundant.

Reply:
Note that `rem_cid` is always set to *something*; that flag identifies whether it's been set to a value determined by the peer, versus the initial random value.

Reply:
The `&self` borrow would render `is_handshaking` ineligible to called while a mutable borrow of one of `Connection`'s fields is still active. This is not currently needed, but in general I like to limit the scope of method borrows to help keep borrowck at bay; for example, this is the sole function of the `Streams` struct.

Reply:
Yeah, I know. But I feel like we may be able to derive that from other bits of state that we have in the relevant places.

Reply:
Alright, I've kept `State::is_handshake()` and made it private.